src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1900,7 +1910,9 @@ private static Binder GetLambdaEnclosingBinder(int position, CSharpSyntaxNode st
         /// </summary>
         protected void EnsureNullabilityAnalysisPerformedIfNecessary()
         {
-            bool isNullableAnalysisEnabled = IsNullableAnalysisEnabled();
+#pragma warning disable RSEXPERIMENTAL001 // internal use of experimental API
+            bool isNullableAnalysisEnabled = !DisableNullableAnalysis && IsNullableAnalysisEnabled();
+#pragma warning restore RSEXPERIMENTAL001
             if (!isNullableAnalysisEnabled && !Compilation.IsNullableAnalysisEnabledAlways)
```

Comment:
>&& !Compilation.IsNullableAnalysisEnabledAlways [](http://example.com/codeflow?start=43&length=47)

When ```DisableNullableAnalysis``` is true, should this condition still matter?

Reply:
I think yes, this condition should remain as written. I will try to break down the expected behaviors, attempting to simplify away details that seem irrelevant. First, enumerating the configuration options which may affect behavior:

- Value of the run-nullable-analysis feature flag: Either "always", "never", or null.
- The "IsNullableAnalysisEnabledInMember" value. I'm "inventing" this value for the purpose of the explanation. The value is true when the combination of `#nullable` directives and `<Nullable>` project-level setting enables nullable analysis within the containing member of the node which the user is asking the semantic model about.
- Compiler build configuration. Either DEBUG or RELEASE.

Next, the expected behaviors with different combinations of these options:
|                                                  | IsNullableAnalysisEnabledInMember=true | IsNullableAnalysisEnabledInMember=false |
|--------------------------------------------------|----------------------------------------|-----------------------------------------|
| run-nullable-analysis=always                     | Run analysis and use results           | Run analysis and ignore results         |
| run-nullable-analysis=never                      | Do not run analysis                    | Do not run analysis                     |
| run-nullable-analysis=null,Configuration=DEBUG   | Run analysis and use results           | Run analysis and ignore results         |
| run-nullable-analysis=null,Configuration=RELEASE | Run analysis and use results           | Do not run analysis                     |

The logic in this if-statement allows us to behave correctly when **"IsNullableAnalysisEnabledInMember"** is **false**, **run-nullable-analysis** is **null**, and **Configuration** is **DEBUG**. If we, for example, simplified the condition on this line to just `!isNullableAnalysisEnabled`, then we would change behavior from "Run analysis and ignore results" to "Do not run analysis".

Hopefully this explanation makes things more clear, I found following the logic here quite difficult myself.

Reply:
> Hopefully this explanation makes things more clear, I found following the logic here quite difficult myself.

Just to confirm, we do want (i.e. it is our intent) to run nullable analysis for nullable disabled semantic model depending on the the run-nullable-analysis feature flag. Is this correct? If so, consider adding an explicit comment regarding this.